# Python Arithmetic Operators --->Python Aritmetik Operatörleri


## Toplama Operatoru--> Addition Operator

Artı olarak telaffuz edilen bu operatör, temel bir aritmetik operatördür. 

Her iki taraftaki iki sayısal işleneni toplar ve toplama sonucunu döndürür.

In [1]:
a =10
b =20
c =a + b
print(c)

30


### Tamsayı ve float  toplaminda sonuc float verir

In [1]:
a =10
b= 20.5
print(a + b)

30.5


## Cikarma Operatoru ---> Subtraction Operator (-)

In [3]:
a =10
b =5 
print( a- b)

5


## Int() ile float() cikarma isleminde sonuc float olur.

In [2]:
a =12
b =5.7
print(a -b)

6.3


## Complex() ve float cikarma isleminde gercek bilesen isleme dahil edilir

In [3]:
a =10 +5j
b =8.5
print(a -b)

(1.5+5j)


## Carpma Operatoru ---> Multiplication Operator (*)

* (yıldız) simgesi Python'da (birçok dilde olduğu gibi) çarpma operatörü olarak tanımlanır. 

Her iki taraftaki iki işlenenin çarpımını döndürür. 

İşlenenlerden herhangi biri negatifse sonuç da negatiftir. 

Her ikisi de negatif ise sonuç pozitiftir. 

İşlenenlerin sırasını değiştirmek sonucu değiştirmez

In [6]:
a =10
b= 20
print( a * b)


200


## carpma isleminde, int ile float  ve bilimsel bir sayi carpilabilir

In [7]:
a = 10
b =20.5
print(a * b)

205.0


In [8]:
a =5.55
b= 6.75E-3
print(a * b)

0.037462499999999996


# complex sayi ile  float sayi carpiminda sonucta complex sayinin sanal kismini da carpar

In [4]:
a =10 + 5j
b = 20.5
print (a * b) ## (10*20.5) ve  (5j * 20.5) carpilir


(205+102.5j)


## Bolme Operatoru ---->  Division Operator (/)
 * islemde herhagi bir sayinin negatif olmasi durumunda sonuc neghatiftir.

 ** Her iki sayi int olsa bile bolme de sonuc floattir.

In [5]:
a =20
b =10
c = (a/b)
print(c)
type(c)

2.0


float

Bolme isleminde float ile tam sayi bolunebilir,veya bilimsel bir sayi da olablilr.

In [6]:
a =10
b = -20.5
c= a/b
print(c)

-0.4878048780487805


In [16]:
a = -2.5
b =125.0
print(a/b)

-0.02


* Complex sayi  ile bolme isleminde complex sayinin her iki kismiyla bolme islemi yapilir.

In [17]:
a =7.5 + 7.5j
b =2.5
c =a/b
print(c)

(3+3j)


## Modul Operatoru ---> Modulus Operator (%)

Python, Yüzde sembolü olarak bilinen "%" sembolünü Modül (veya modülo) operatörü olarak tanımlar.

Payda payı böldükten sonra kalanı döndürür. Kalan operatörü olarak da adlandırılabilir.  

Modül operatörünün sonucu, tamsayı bölümünden sonra kalan sayıdır.
 
Örnek vermek gerekirse, 10, 3'e bölündüğünde bölüm 3, kalan 1 olur. Dolayısıyla %103 (normalde 10 mod 3 olarak telaffuz edilir) 1 sonucunu verir.

Her iki işlenen de tam sayı ise modül değeri bir tam sayıdır.
Pay tamamen bölünebilirse kalan 0'dır.

Pay, paydadan küçükse modül paya eşittir. Payda 0 ise Python ZeroDivisionError 
değerini yükseltir.

In [7]:
a =10
b =2
print(a % b)

0


In [19]:
a = 10
b = 4
print(a % b)

2


In [20]:
a = 0
b = 10
print(a % b)

0


* mod  alma isleminde sayilardan herhangi biri float ise sonuc float olur.

* mod alma isleminde complex sayiyi kabul etmrz .Hata verir.

In [21]:
a =10
b = 2.5
print(a % b)

0.0


In [22]:
a =10
b =1.5
print(a % b)

1.0


# Python Us Operatoru(**)---> Exponent Operator (**)

* Python üs operatörü olarak ** (çift yıldız işareti) kullanır.

* Üs alma ifadesinde her iki işlenen de tam sayı ise sonuç da bir tam sayıdır. 

*Her ikisinin de float durumunda float  olur.

* Sayilardan biri complex sayi ise us operatoru complex sayi dondurur.

* Taban  0 ise sonuc 0 , index 0 ise  sonuc her zaman 1 olur.


### pow() ---> us alma fonksiyonu


In [8]:
a =10
b = 2
print( a ** b)
type(a ** b)

100


int

In [9]:
a =10
b =1.5
print( a ** b)
type (a ** b)

31.622776601683793


float

In [10]:
a =1+2j
b =4
print( a ** b)
type(a ** b)

(-7-24j)


complex

In [28]:
a =12.4
b = 0
print( a** b)

1.0


In [2]:
25 ** 0.5

5.0

In [11]:
pow(13,3)

2197

In [44]:
pow(3,2)

125


In [ ]:
x =(4)

# Cift Bolme Operatoru(//)---> Floor Division Operator (//)

* Kat bölmesine tam sayı bölmesi de denir. Python bu amaçla // (çift eğik çizgi) sembolünü kullanır. Geri kalanı döndüren modül veya modülün aksine, taban bölümü, ilgili işlenenlerin bölümünün bölümünü verir.

* Her iki sayi da pozitif ise kat operatoru kesirli kismi cikarilmis bir sayi dondurur.

* Ancak iki sayidan biri negatif ise sonuc sifirdan uzaklasarak negatf sonsuza yuvarlanir.


In [12]:
a =9
b =2 
print(a // b)

4


In [13]:
a =9
b = -2
print(a // b)

-5


In [14]:
a = 10
b =1.5
print( a // b)

6.0


In [33]:
a = -10
b= 1.5
print( a//b)

-7.0


In [21]:
y = 7    # 3 defa calistirlimnca sonuc ne olur?
y += 5
print(y)
# her seferinde y degiskeninin yeniden ayni deger olrak tanimlar.
# Bu yuzden sonuc degismez.

12


In [18]:
    # 3 kere calistirilinca sonuc ne olur?
y += 5
print(y)

27


SORU : Bir kenari 5 diger kenari 12 olan dik ucgenin hipotenusu kactir?

In [22]:

 
kisa_kenar =  int(input("kisa kenari giriniz"))
uzun_kenar = int(input("uzunkenari giriniz"))
hipotenus = (kisa_kenar ** 2 + uzun_kenar ** 2) ** 0.5
print("hipotenus:", hipotenus)


hipotenus: 13.0


# Python Complex Sayi Aritmetigi ----> Complex Number Arithmetic

* Aritmetik operatörler, her iki sayi da complex sayi  olduğunda biraz farklı davranır.

* Complex  sayılarda toplama ve çıkarma işlemi, ilgili gerçek ve sanal bileşenlerin basit bir şekilde toplanması/çıkarılmasıdır.

In [23]:
a =2.5+3.4j
b = 3+1.0j
print(a + b)
print(a -b)

(5.5+4.4j)
(-0.5+2.4j)


Karmaşık sayıların çarpımı cebirdeki iki binomun çarpımına benzer.

 Eğer "a+bj" ve "x+yj" iki karmaşık sayı ise bunların çarpımı şu formülle verilir:

 (a+bj)*(x+yj) = ax+ayj+xbj+byj2 = (ax-by)+(ay+xb)j

In [ ]:
a=6+4j
b=3+2j
c=a*b
c=(18-8)+(12+12)j
c=10+24j

İki complex  sayının bölünmesinin nasıl gerçekleştiğini anlamak için karmaşık sayının eşlenikini kullanmalıyız.

Python'un karmaşık nesnesi, sanal kısmın işareti ters çevrilmiş bir karmaşık sayı döndüren bir conjugate() yöntemine sahiptir.

In [38]:
a =5+6j
a.conjugate()

(5-6j)

İki complex  sayıyı bölmek için, hem payı hem de paydayı paydanın eşleniğiyle bölün ve çarpın.

In [41]:
a=6+4j
b=3+2j
c=a/b
c=(6+4j)/(3+2j)
c=(6+4j)*(3-2j)/3+2j)*(3-2j)
c=(18-12j+12j+8)/(9-6j+6j+4)
c=26/13
c=2+0j



SyntaxError: unmatched ')' (1225027925.py, line 5)

In [42]:
a =6+4j
b = 3+2j
print(a /b)

(2+0j)


Python'daki complex  sınıf, modül operatörünü (%) ve kat bölme operatörünü (//) desteklemez.

soru = yaricapi 7 olan dairenin alanini bulan python kodunu yaziniz

In [6]:
r = 7
pi =3.14

alan = pi * r **2
print("Dairenin alani:", alan)

Dairenin alani: 153.86


soru = y =(2 * 6 /4) ** (5 * (21 // 4)- 23)
print (y) sonuc nedir

In [7]:
y =(2 * 6 /4) ** (5 *(21 //4)-23)
print(y)

9.0


soru : Kullanicidan alinan iki sayi alin ve tum aritmetik islem operatorlerinin  kullanarak sonuclari ekrana yazdirin.

In [3]:
num1 = float(input("1.sayiyi giriniz"))
num2 = float(input("2.sayiyi giriniz"))
print ("Toplama sonucu:",num1 + num2)
print("cikarma sonucu:",num1 - num2 )
print("carpim sonucu :",num1 * num2)
print("bolme sonucu :", num1 / num2)
print("ussu alma:",num1 ** num2)
print("modulus sonucu:", num1 % num2 )
# ilk sayi =5.2
# 2.sayi 5.3

Toplama sonucu: 10.5
cikarma sonucu: -0.09999999999999964
carpim sonucu : 27.56
bolme sonucu : 0.9811320754716982
ussu alma: 6234.73101230862
modulus sonucu: 5.2


soru : Kullanicidan once adini,sonra soyadini ve daha sonra yasini isteyiniz.
Alinan sonucu tek bir 

In [5]:
name = input("Adiniz: ")
lastname = input("Soyadiniz :")
age = input("Yas :")
print(name,lastname,age)
print(name,lastname,age,sep ='\n')

ali candan 40
ali
candan
40


soru : Kullanicidan iki sayi isteyin.1 sayinin kuvveti olarak 2.sayiyi alin,elde edilen sonucun 4 e bolumunden kalan sayiyi ekrana yazdirin.

In [6]:
sayi1 =  int(input("sayi1"))
sayi2 =int(input("sayi2:"))
elde_edilen_sayi = sayi1 ** sayi2
sonuc = elde_edilen_sayi % 4 
print("sonuc:",sonuc)
# sayi1 =5
#sayi2 =4

sonuc: 1


soru : Kullanicidan iki sayi isteyin.1.sayiyi integer veri tipine,2.sayiyi Float veri tipine donusturun.Aldiginiz 2 sayinin farkini yeni bir degiskene aktarin ve bu degiskeni ekrana bastirin.
daha sonra bu degiskeni Integer veri tipine donusturerek ekrana yazdirin.

In [7]:
sayi_1 = int((input("sayi 1 :")))
sayi_2 = float(input("sayi 2 :"))
result = sayi_1 -sayi_2
print(result)
result =int(result)
print(result)

# sayi_1 =12
#sayi_2 =4.7

7.3
7


In [8]:
x =(4 // 7) ** (4 **(1 * 2 / 2)/2)
print(x)

0.0


In [9]:
25 % 7

4

In [10]:
x =(24 /2) - (2 ** (1 * 8 /2)/2)
print(x)

4.0


In [11]:
25 // 7

3

In [12]:
25 //7.0

3.0

In [16]:
z =5.0 # 3 kere calistir sonuc ne olur?
z *= 3
print(z)

15.0


soru: Kullanicidan celcius bilgisini alin ve fahrenheit a cevirin

In [18]:
celc = float(input("hava sicakligini celcius olarak yaziniz.Orn:25.5"))
fahr = celc * 1.8 + 32

print("Hava sicakligi",celc,"derece:",fahr)

Hava sicakligi 25.5 derece: 77.9
